frontend/resourceadm/components/NewResourceModal/NewResourceModal.tsx

```diff
@@ -1,162 +1,115 @@
-import React, { useState } from 'react';
-import classes from './NewResourceModal.module.css';
-import { Button } from '@digdir/design-system-react';
-import { Modal } from '../Modal';
+import React, { forwardRef, useState } from 'react';
+import { Button, Paragraph, Modal } from '@digdir/design-system-react';
 import { ResourceNameAndId } from '../ResourceNameAndId';
 import { useCreateResourceMutation } from 'resourceadm/hooks/mutations';
 import { useNavigate } from 'react-router-dom';
 import type { NewResource } from 'app-shared/types/ResourceAdm';
 import { getResourcePageURL } from 'resourceadm/utils/urlUtils';
 import { useTranslation } from 'react-i18next';
-import { replaceWhiteSpaceWithHyphens } from 'resourceadm/utils/stringUtils';
 import { ServerCodes } from 'app-shared/enums/ServerCodes';
 import { useUrlParams } from 'resourceadm/hooks/useSelectedContext';
 
 export type NewResourceModalProps = {
-  isOpen: boolean;
   onClose: () => void;
 };
 
 /**
  * @component
  *    Displays the modal telling the user that there is a merge conflict
  *
- * @property {boolean}[isOpen] - Boolean for if the modal is open
  * @property {function}[onClose] - Function to handle close
  *
- * @returns {React.ReactNode} - The rendered component
+ * @returns {JSX.Element} - The rendered component
```

Comment:
Great that you refactor this! 
The ```React.ReactNode``` is something I did many months ago before realising it is better to use ```JSX.Element```, so there might be a few places where this should be changed throughout the ```resoureadm``` code 😄 
I do not think all should be changed in this PR, as it is way too many places, but I like that you change it when you work in the file 👏